task_sdk/src/airflow/sdk/definitions/dag.py

```diff
@@ -557,9 +557,9 @@ def __lt__(self, other):
     def __hash__(self):
         hash_components: list[Any] = [type(self)]
         for c in _DAG_HASH_ATTRS:
-            # task_ids returns a list and lists can't be hashed
-            if c == "task_ids":
-                val = tuple(self.task_dict)
+            # If it is a list, convert to tuple because lists can't be hashed
```

Comment:
@ashb I made this more generic since `watchers` is also a list and I thought we could just say "if it is a list, then let's convert it to a tuple". Let me know if you have any concerns